tests/integration/test_k8s.py

```diff
@@ -136,21 +136,32 @@ async def test_remove_worker(kubernetes_cluster: model.Model):
 
 
 @pytest.mark.abort_on_fail
-async def test_remove_non_leader_control_plane(kubernetes_cluster: model.Model):
+async def test_remove_non_leader_control_planes(kubernetes_cluster: model.Model):
     """Deploy the charm and wait for active/idle status."""
     k8s = kubernetes_cluster.applications["k8s"]
     worker = kubernetes_cluster.applications["k8s-worker"]
     expected_nodes = len(k8s.units) + len(worker.units)
     leader_idx = await get_leader(k8s)
     leader = k8s.units[leader_idx]
     follower = k8s.units[(leader_idx + 1) % len(k8s.units)]
+    follower2 = k8s.units[(leader_idx + 2) % len(k8s.units)]
     await ready_nodes(leader, expected_nodes)
 
-    # Remove a control-plane
+    # Remove both non-leading control-planes (Replicate issue #277)
     log.info("Remove unit %s", follower.name)
     await follower.destroy()
     await kubernetes_cluster.wait_for_idle(status="active", timeout=5 * 60)
     await ready_nodes(leader, expected_nodes - 1)
+
+    log.info("Remove unit %s", follower2.name)
+    await follower2.destroy()
+    await kubernetes_cluster.wait_for_idle(status="active", timeout=5 * 60)
+    await ready_nodes(leader, expected_nodes - 2)
+
+    await k8s.add_unit()
+    await kubernetes_cluster.wait_for_idle(status="active", timeout=5 * 60)
+    await ready_nodes(leader, expected_nodes)
```

Comment:
You've added 1 unit, not 2 -- you need to expect only 4 nodes
```suggestion
    await ready_nodes(leader, expected_nodes - 1)
```